Fix OS X framework install path

eyeplum opened the pull request:
Previously, the value of the build setting `INSTALL_PATH` is `$(SYMROOT)/Headers` which may prevent the dynamic linking from working correctly.

Usually, a dynamic linked OS X framework is copied to `APP_BUNDLE/Contents/Frameworks`, while the executable is usually located at `APP_BUNDLE/Content/MacOS/EXECUTABLE_NAME`. As per [apple document](https://developer.apple.com/library/mac/documentation/MacOSX/Conceptual/BPFrameworks/Tasks/CreatingFrameworks.html), the suggested install path of an OS X framework is `@executable_path/../Frameworks`. 

This pull request changes the build setting `INSTALL_PATH` for target `pop-osx-framework` to `@executable_path/../Frameworks`.

eyeplum commented:
@facebook-github-bot CLA signed 😃

nlutsenko commented:
Awesome catch!
Let's use `@rpath` in this case, which is the most flexible option of all.
This change would basically require for `pop` to always be embedded with the application, which might not be the case, where say a framework is shared between two or three applications, every one of which links it differently.

`@rpath` asks the dynamic linker to search a list of locations for the library. That list is embedded in the application, and can be controlled by the application's build process.
This basically means that a single copy of a framework can thus work for multiple purposes.

eyeplum commented:
@nlutsenko Thanks for the review.

I've changed the `INSTALL_PATH` to `@rpath` and have tested in a simple Cocoa app, it works.

Also, it seems that the `PUBLIC_HEADERS_FOLDER_PATH` was set to a folder adjacent to the framework build product, as a result, users of the framework may have to manually copy the header files to use the framework.
I changed the `PUBLIC_HEADERS_FOLDER_PATH` to `$(CONTENTS_FOLDER_PATH)/Headers`, so the header files will be correctly copied to the framework bundle. Thus users can use the framework as-is.

nlutsenko commented:
Yup, that's another amazing catch.
I think we should just move all of the configurations to use our [xctoolchain](https://github.com/parseplatform/xctoolchain) :)

Waiting for build and merging. Thanks again!

eyeplum commented:
🎉 Great! Glad I can help!
